sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+    uint32_t Timeout = uint32_t();  // The timeout parameter is expressed in seconds. For more information, see <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting Timeouts for Blob Service Operations.</a>
+    std::string ApiVersionParameter = k_DefaultServiceApiVersion;  // Specifies the version of the operation to use for this request.
+    PathLeaseAction XMsLeaseAction;  // There are five lease actions: "acquire", "break", "change", "renew", and "release". Use "acquire" and specify the "x-ms-proposed-lease-id" and "x-ms-lease-duration" to acquire a new lease. Use "break" to break an existing lease. When a lease is broken, the lease break period is allowed to elapse, during which time no lease operation except break and release can be performed on the file. When a lease is successfully broken, the response indicates the interval in seconds until a new lease can be acquired. Use "change" and specify the current lease ID in "x-ms-lease-id" and the new lease ID in "x-ms-proposed-lease-id" to change the lease ID of an active lease. Use "renew" and specify the "x-ms-lease-id" to renew an existing lease. Use "release" and specify the "x-ms-lease-id" to release a lease.
+    uint64_t XMsLeaseDuration;  // The lease duration is required to acquire a lease, and specifies the duration of the lease in seconds.  The lease duration must be between 15 and 60 seconds or -1 for infinite lease.
+    uint64_t XMsLeaseBreakPeriod;  // The lease break period duration is optional to break a lease, and  specifies the break period of the lease in seconds.  The lease break  duration must be between 0 and 60 seconds.
+    std::string LeaseIdOptional = std::string();  // If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+    std::string ProposedLeaseIdOptional = std::string();  // Proposed lease ID, in a GUID string format. The Blob service returns 400 (Invalid request) if the proposed lease ID is not in the correct format. See Guid Constructor (String) for a list of valid GUID string formats.
+    std::string IfMatch = std::string();  // Specify an ETag value to operate only on blobs with a matching value.
+    std::string IfNoneMatch = std::string();  // Specify an ETag value to operate only on blobs without a matching value.
+    std::string IfModifiedSince = std::string();  // Specify this header value to operate only on a blob if it has been modified since the specified date/time.
+    std::string IfUnmodifiedSince = std::string();  // Specify this header value to operate only on a blob if it has not been modified since the specified date/time.
+  };
+
+  static Azure::Core::Http::Request PathLeaseCreateRequest(std::string url, const PathLeaseOptions& pathLeaseOptions)
```

Comment:
```suggestion
  static Azure::Core::Http::Request PathLeaseCreateRequest(const std::string& url, const PathLeaseOptions& pathLeaseOptions)
```

Reply:
The request URL is changed to moved, I think it makes more sense for `Request` to have `url` accepting rvalue/by value as well